test/shape_inference.jl

```diff
@@ -194,6 +194,21 @@ end
     end
 end
 
+
+@testset "dropout" begin
+    for var in (m,n,k)
+        @test get_shape(nn.dropout(var, 0.5)) == get_shape(var)
+    end
+
+
+    m_by_i = nn.dropout(m, 0.5*i) # a scalar keep_prob
+    @test get_shape(m_by_i) == get_shape(m)
+
+    m_by_n = nn.dropout(m, 0.5*n) # a fully unknown keepprob
+    @test get_shape(m_by_n).rank_unknown == true
```

Comment:
Is this just a limitation of the current shape inference code, or the truly desired behavior?